src/site/markdown/server/manage-guice-distributed-james.md

```diff
@@ -260,3 +261,70 @@ by setting the parameter `elasticsearch.index.mailbox.name` to the name of your
 re-creates index upon restart
 
 _Note_: keep in mind that reindexing can be a very long operation depending on the volume of mails you have stored.
+
+## Solving cassandra inconsistencies
+
+By choosing the cassandra, it means denormalization is the common modeling pattern to be using. 
+You may see some data is duplicated among tables. Especially, with tables having close relationships. 
+Some operations require multiple table writes, if there is any failure upon in the middle of an operation. 
+It could bring an inconsistent state to the data. The consequence could be dirty reads, unexpected failing writes.
+
+Because of the lack of transactions, it's hard to prevent these kind of issues. We had developed some features to 
+fix some existing cassandra inconsistency issues that had been reported to James. 
+
+There are typical type inconsistencies:
+ - [RRT (RecipientRewriteTable) mapping sources](#rrt-recipientrewritetable-mapping-sources)
+ - [Jmap message fast view projections](#jmap-message-fast-view-projections)
+ - [Mailboxes](#mailboxes)
+
+### RRT (RecipientRewriteTable) mapping sources
+
+`rrt` and `mappings_sources` tables store information about address mappings. 
+While `rrt` is the source of truth and `mappings_sources` is the projection table containing all 
```

Comment:
While?

Reply:
do you think there is anything wrong?

Reply:
sorry, my comment was not clear. It wonder why you start the sentence with "While"